test/store.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+CURRENT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+if [[ -f "${CURRENT_DIR}/store-port" ]]; then
+    export STORE_PORT="$(cat ${CURRENT_DIR}/store-port)"
+fi
+
+# store_init initialises a store and waits for it to come online.
+# store_init also stores the port the store is listening in "${CURRENT_DIR}/store-port"
+function store_init() {
+  rm -rf "${CURRENT_DIR}/store-logs" "${CURRENT_DIR}/store-port"
+
+  go build -o "${CURRENT_DIR}/store" "${CURRENT_DIR}/kv/cmd"
+  "${CURRENT_DIR}/store" serve &> "${CURRENT_DIR}/store-logs" &
+
+  sleep 1
+  cat "${CURRENT_DIR}/store-logs" | \
+   grep "Using port:" | \
+    awk '{ print $5 }' | \
+     xargs echo -n > "${CURRENT_DIR}/store-port"
+  export STORE_PORT="$(cat ${CURRENT_DIR}/store-port)"
```

Comment:
Would it be more straight-forward to get the port value, and then writing it to a file? (Rather than writing to a file and reading back)
```
  export STORE_PORT=$(cat "${CURRENT_DIR}/store-logs" | \
   grep "Using port:" | \
    awk '{ print $5 }')
  echo $STORE_PORT > "${CURRENT_DIR}/store-port"
```